libsplinter/src/admin/service/shared.rs

```diff
@@ -1648,6 +1648,43 @@ impl AdminServiceShared {
         service_id: &str,
         protocol: u32,
     ) -> Result<(), AdminSharedError> {
+        // Update any unpeered payloads that this service might be a member of
+        let mut unpeered_payloads = std::mem::replace(&mut self.unpeered_payloads, vec![]);
+        for pending_protocol_payload in unpeered_payloads.iter_mut() {
+            match protocol {
+                0 => {
+                    if pending_protocol_payload
+                        .missing_protocol_ids
+                        .iter()
+                        .any(|missing_protocol_id| missing_protocol_id == service_id)
+                    {
+                        warn!(
+                            "Dropping circuit request including service {}, \
+                             due to protocol mismatch",
```

Comment:
Add the mismatched protocol versions to this message for debugging purposes.

Reply:
It's actually in a message in the caller of this function.  Any bad protocol at the call site is translated to `0`